include/fdreadoutlibs/wib/RAWWIBTriggerPrimitiveProcessor.hpp

```diff
@@ -80,175 +80,37 @@ class RAWWIBTriggerPrimitiveProcessor
-TLOG_DEBUG(TLVL_WORK_STEPS) << "Number of hits, padding " << debug_nhits << ", " << debug_padding;
-
-// raw recording 
-tp_dump_frame();
-
-// stitch TP hits
-tp_stitch(std::move(&m_payload_wrapper));
-}
-
-void unpack_tpframe_version_2(frame_ptr fr, int& nhits, int& offset)
-{
-  auto& srcbuffer = fr->get_raw_tp_frame_chunk();
-
-  dunedaq::detdataformats::wib::RawWIBTp* rwtps =
-         static_cast<dunedaq::detdataformats::wib::RawWIBTp*>( malloc(
-         sizeof(dunedaq::detdataformats::wib::TpHeader) + nhits * sizeof(dunedaq::detdataformats::wib::TpData)
-         ));
-  m_payload_wrapper.rwtp.release( );
-  m_payload_wrapper.rwtp.reset( rwtps );
-
-  m_payload_wrapper.rwtp->set_nhits(nhits);
-
-  ::memcpy(static_cast<void*>(&m_payload_wrapper.rwtp->m_head),
-           static_cast<void*>(srcbuffer.data() + offset),
-           2*RAW_WIB_TP_SUBFRAME_SIZE);
-
-  for (int i=0; i<nhits; i++) {
-    ::memcpy(static_cast<void*>(&m_payload_wrapper.rwtp->m_blocks[i]),
-             static_cast<void*>(srcbuffer.data() + offset + (2+i)*RAW_WIB_TP_SUBFRAME_SIZE),
-             RAW_WIB_TP_SUBFRAME_SIZE);
-  }
-
-  int debug_nhits = m_payload_wrapper.rwtp->get_nhits();
-  uint16_t debug_padding = m_payload_wrapper.rwtp->get_padding_3(); // NOLINT
-  //TLOG() << "Number of hits, padding " << debug_nhits << ", " << debug_padding; 
-  TLOG_DEBUG(TLVL_WORK_STEPS) << "Number of hits, padding " << debug_nhits << ", " << debug_padding;
-
-  // raw recording 
-  tp_dump_frame();
-
-  // stitch TP hits
-  tp_stitch(std::move(&m_payload_wrapper));
-}
-
-void tp_unpack(frame_ptr fr)
-{
-  auto& srcbuffer = fr->get_raw_tp_frame_chunk();
-  int num_elem = fr->get_raw_tp_frame_chunksize();
-
-  if (num_elem == 0) {
-    TLOG_DEBUG(TLVL_WORK_STEPS) << "No raw WIB TP elements to read from buffer! ";
-    return;
-  }
-  if (num_elem % RAW_WIB_TP_SUBFRAME_SIZE != 0) {
-    TLOG_DEBUG(TLVL_WORK_STEPS) << "Raw WIB TP elements not multiple of subframe size (3)! ";
-    return;
-  }
-
-  int offset = 0;
-  while (offset <= num_elem) {
-
-    if (offset == num_elem) break;
-
-    // Create next TP frame
-    m_payload_wrapper.rwtp.reset(new types::RAW_WIB_TRIGGERPRIMITIVE_STRUCT::FrameType());
-    ::memcpy(static_cast<void*>(&m_payload_wrapper.rwtp->m_head),
-             static_cast<void*>(srcbuffer.data() + offset),
-             m_payload_wrapper.rwtp->get_header_size());
-    int nhits = m_payload_wrapper.rwtp->get_nhits();
-    uint16_t padding = m_payload_wrapper.rwtp->get_padding_3(); // NOLINT
-
-    if (padding == 48879) { // padding hex is BEEF, new TP format
-      TLOG_DEBUG(TLVL_WORK_STEPS) << "Unpack raw WIB TP frame version 2.";
-      unpack_tpframe_version_2(fr, nhits, offset);
-
-    } else { // old TP format
-      TLOG_DEBUG(TLVL_WORK_STEPS) << "Unpack raw WIB TP frame version 1.";
-      unpack_tpframe_version_1(fr, offset);
+      fwrite(&rwtp->m_head, 1, sizeof(dunedaq::detdataformats::wib::TpHeader),
```

Comment:
I would appreciate to avoid using C fwrite. This is (mainly) C++ code, and we should only use native C code when necessary. Please move to fstream.

Reply:
@hristovaivana I think this is resolved.  If so, can you resolve the conversation.